sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -131,31 +138,43 @@ class OrcFileFormat extends FileFormat with DataSourceRegister with Serializable
         val physicalSchema = maybePhysicalSchema.get
         OrcRelation.setRequiredColumns(conf, physicalSchema, requiredSchema)
 
-        val orcRecordReader = {
-          val job = Job.getInstance(conf)
-          FileInputFormat.setInputPaths(job, file.filePath)
-
-          val fileSplit = new FileSplit(
-            new Path(new URI(file.filePath)), file.start, file.length, Array.empty
-          )
-          // Custom OrcRecordReader is used to get
-          // ObjectInspector during recordReader creation itself and can
-          // avoid NameNode call in unwrapOrcStructs per file.
-          // Specifically would be helpful for partitioned datasets.
-          val orcReader = OrcFile.createReader(
-            new Path(new URI(file.filePath)), OrcFile.readerOptions(conf))
-          new SparkOrcNewRecordReader(orcReader, conf, fileSplit.getStart, fileSplit.getLength)
+        val job = Job.getInstance(conf)
+        FileInputFormat.setInputPaths(job, file.filePath)
+
+        val fileSplit = new FileSplit(
+          new Path(new URI(file.filePath)), file.start, file.length, Array.empty
+        )
+        // Custom OrcRecordReader is used to get
+        // ObjectInspector during recordReader creation itself and can
+        // avoid NameNode call in unwrapOrcStructs per file.
+        // Specifically would be helpful for partitioned datasets.
+        val orcReader = OrcFile.createReader(
+          new Path(new URI(file.filePath)), OrcFile.readerOptions(conf))
+
+        if (enableVectorizedReader) {
+          val conf = job.getConfiguration.asInstanceOf[JobConf]
```

Comment:
why can't you reuse the `conf` at line 129 (https://github.com/apache/spark/pull/13775/files#diff-01999ccbf13e95a0ea2d223f69d8ae23R129) ?